### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by moinnadeem**
Adding revised README with Jonathan's feedback, and improved algorithm tables.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,81 +37,126 @@
...
+* We provide <b>curated recipes of speedup methods</b> that lead to the most efficient possible training on these benchmarks.
 
-Composer features:
-- 20+ efficient training methods for training a better language and vision models! Don't waste hours trying to reproduce research papers when Composer has done the work for you.
-- Easy-to-use (optional) Trainer interface written to be as performant as possible, and [integrated best practices](https://www.mosaicml.com/blog/5-best-practices-for-efficient-model-training).
-- Easy-to-use Functional forms that allow you to integrate efficient training methods into your training loop!
-- Strong, _reproducible_ baselines to get you started as 💨 fast 💨 as possible
 
-## Quickstart
+## 📣 Quickstart
```
Comment (ravi-mosaicml) [inline README.md:80]: Can we move the quickstart (or at least the installation section) up higher? I think it would be helpful if it is above the fold

File: README.md [hunk 2]
```diff
@@ -34,81 +37,126 @@
...
 
 ### Usage
 
-Composer provides both a **Functional API** (similar to `torch.nn.functional`) and a **Trainer** (that abstracts away the training loop) to provide flexibility to users.
+You can use Composer's speedup methods in two ways:
+
+* Through a standalone **Functional API** (similar to `torch.nn.functional`) that allows you to integrate them into your existing training code.
+* Using Composer's built-in **Trainer**, which is designed to be performant and automatically takes care of many of the low-level details of using speedup methods.
 
 #### Example: Functional API [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/mosaicml/composer/blob/dev/notebooks/Composer_Functional.ipynb)
```
Comment (ravi-mosaicml) [inline README.md:102]: maybe move the "open in colab" button to a new line, immediately below the header?

File: README.md [hunk 3]
```diff
@@ -34,81 +37,126 @@
...
 
-For users who choose to use their own training loop, we provide state-less functional implementations of our algorithms for a end-user to integrate.
+For users who wish to integrate Composer's speedup methods into their existing training code, we provide stateless functional implementations of these methods.
+
+The following example highlights using [BlurPool](https://docs.mosaicml.com/en/latest/method_cards/blurpool.html), which applies an anti-aliasing filter before every downsampling operation in convolutional neural networks like ResNet.
 
-The following example highlights using [BlurPool](https://arxiv.org/abs/1904.11486), which applies an anti-aliasing filter before every downsampling operation.
+TODO(MOIN): Update the below example to make sure (a) it can run on its own and (b) it matches the MNIST example below in terms of the training setup so that the two are really equivalent. Also, both examples should use both blurpool and label smoothing.
```
Comment (ravi-mosaicml) [inline README.md:108]: Creating a comment to track the TODO...

File: README.md [hunk 4]
```diff
@@ -34,81 +37,126 @@
...
+        smoothed_targets = cf.smooth_labels(outputs, labels, smoothing=0.1)
         loss = criterion(outputs, labels)
         loss.backward()
         optimizer.step()
+    scheduler.step()
 ```
 
 See the official [Composer Functional API Colab notebook](https://colab.research.google.com/github/mosaicml/composer/blob/dev/notebooks/Composer_Functional.ipynb) for more.
 
 #### Example: Trainer [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/mosaicml/composer/blob/dev/notebooks/up_and_running_with_composer.ipynb)
```
Comment (ravi-mosaicml) [inline README.md:157]: same comment re: moving the open in colab button below.

File: README.md [hunk 5]
```diff
@@ -34,81 +37,126 @@
...
-        outputs = net(inputs)
+        outputs = model(inputs)
+        smoothed_targets = cf.smooth_labels(outputs, labels, smoothing=0.1)
         loss = criterion(outputs, labels)
         loss.backward()
         optimizer.step()
+    scheduler.step()
 ```
 
 See the official [Composer Functional API Colab notebook](https://colab.research.google.com/github/mosaicml/composer/blob/dev/notebooks/Composer_Functional.ipynb) for more.
```
Comment (ravi-mosaicml) [inline README.md:155]:
```suggestion
See the [Composer Functional API Colab notebook](https://colab.research.google.com/github/mosaicml/composer/blob/dev/notebooks/Composer_Functional.ipynb) for more.
```

File: README.md [hunk 6]
```diff
@@ -34,81 +37,126 @@
...
 
 #### Example: Trainer [![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/mosaicml/composer/blob/dev/notebooks/up_and_running_with_composer.ipynb)
 
-For maximal speedups, we recommend using our Trainer, which manages handling user state, performant algorithm implementations, and provides useful engineering abstractions to permit rapid experimentation.
+For the best experience and the most efficient possible training, we recommend using Composer's built-in trainer, which automatically takes care of the low-level details of using speedup methods and provides useful abstractions that facilitate rapid experimentation.
```
Comment (ravi-mosaicml) [inline README.md:159]:
```suggestion
For the best experience and the most efficient possible training, we recommend using [Composer's built-in trainer](link to trainer docs), which automatically takes care of the low-level details of using speedup methods and provides useful abstractions that facilitate rapid experimentation.
```

File: README.md [hunk 7]
```diff
@@ -120,71 +168,100 @@ from torchvision import datasets, transforms
...
-
-MosaicML uses a _benchmark_ as a term to denote an reproducible standard within the machine learning community. A benchmark is a specific model trained for a task, where a task is defined as a specific dataset with a specific loss function.
-
-Composer is currently focused on supporting computer vision and natural language processing use cases. We currently support the following combinations of models, datasets, and loss functions.
+
+Composer's built-in trainer makes it easy to **add multiple speedup methods in a single line of code!** 
+Trying out new methods or combinations of methods is as easy as changing a single list.
+As we continually implement more methods, they will be easy for you to add to your code.
+
+For concrete examples of methods in Composer, here are some ([_see here for all_](https://docs.mosaicml.com/en/latest/method_cards/methods_overview.html)) speedup methods currently in Composer:
```
Comment (ravi-mosaicml) [inline README.md:207]: Curious why we are not showing the complete list by default? In addition, maybe rephrase the all link with "A complete list of methods is available [here]"?

File: README.md [hunk 8]
```diff
@@ -120,71 +168,100 @@ from torchvision import datasets, transforms
...
+They may optimize the [same part of the training process](https://en.wikipedia.org/wiki/Amdahl's_law) and lead to diminishing returns, or they may even interact in ways that prove detrimental.
+Determining which methods to compose together isn't as simple as assembling a set of methods that perform best individually.
+
+**We have come up with compositions of methods that work especially well together** through rigorous exploration of the design space of recipes and research on the science behind composition.
+The [MosaicML Explorer](https://app.mosaicml.com/) contains all of the data we have collected so far on composition, and it highlights the compositions of methods that are _pareto-optimal_ - that provide the **best possible tradeoffs between training time or cost and the quality of the trained model**.
+Whether you want to reach the same quality faster or get better quality within your current budget, Explorer can help you decide which speedup methods to use.
+We update this data regularly as we add new methods and develop better recipes.
+
+<p align="center">
+<img src="https://storage.googleapis.com/docs.mosaicml.com/images/methods/explorer.png"/>
```
Comment (ravi-mosaicml) [inline README.md:235]: This image is low resolution on a 4k monitor. Can we re-screenshot at a higher resolution?

File: README.md [hunk 9]
```diff
@@ -120,71 +168,100 @@ from torchvision import datasets, transforms
...
+[MixUp](https://github.com/mosaicml/composer/tree/dev/composer/algorithms/mixup)|`CF.mixup_batch`|[Blends pairs of examples and labels.](https://arxiv.org/abs/1710.09412)|ResNet-101|1.5x
+[Progressive Resizing](https://github.com/mosaicml/composer/tree/dev/composer/algorithms/progressive_resizing)|`cf.resize_batch`|[Increases the input image size during training.](https://github.com/fastai/fastbook/blob/780b76bef3127ce5b64f8230fce60e915a7e0735/07_sizing_and_tta.ipynb)|ResNet-101|1.3x
+[SAM](https://github.com/mosaicml/composer/tree/dev/composer/algorithms/sam)|`N/A`|[SAM optimizer measures sharpness of optimization space.](https://arxiv.org/abs/2010.01412)|ResNet-101|1.5x
+**Composition** | `N/A` | **Cheapest: $49 @ 78.1% Acc** | **ResNet-101** | **3.5x**
+
+#### GPT-2
+
+Name|Functional|tl;dr|Benchmark|Speed Up
+----|----------|-----|---------|--------
+[Alibi](https://github.com/mosaicml/composer/tree/dev/composer/algorithms/alibi)|`cf.apply_alibi`|[Replace attention with AliBi](https://arxiv.org/abs/2108.12409v1)|GPT-2|1.6x
```
Comment (ravi-mosaicml) [inline README.md:256]:
```suggestion
[Alibi](https://github.com/mosaicml/composer/tree/dev/composer/algorithms/alibi)|`cf.apply_alibi`|[Replace attention with AliBi.](https://arxiv.org/abs/2108.12409v1)|GPT-2|1.6x
```

File: README.md [hunk 10]
```diff
@@ -120,71 +168,100 @@ from torchvision import datasets, transforms
...
+[Seq Length Warmup](https://github.com/mosaicml/composer/tree/dev/composer/algorithms/seq_length_warmup)|`cf.set_batch_`<br>`sequence_length`|[Progressively increase sequence length.](https://arxiv.org/abs/2108.06084)|GPT-2|1.5x
+**Composition** | `N/A` | **Cheapest: $150 @ 24.11 PPL** | **GPT-2** | **1.7x**
+
+## ⚙️ What benchmarks does Composer support?
+
+Composer uses a _benchmark_ as a term to denote a particular model trained on a particular dataset in a standardized, reproducible way. A benchmark is a specific model trained for a task, where a task is defined as a dataset with a specific loss function.
```
Comment (ravi-mosaicml) [inline README.md:262]: This is a bit verbose.

```suggestion
A _benchmark_ is a particular model trained on a particular dataset with a specific loss function.
```

I think it is implied that a benchmark is reproducible?

Also, is scheduling part of a benchmark?

File: README.md [hunk 11]
```diff
@@ -257,33 +334,36 @@ Composer is currently focused on supporting computer vision and natural language
...
-
-Furthermore, combining these efficiency methods together isn't a piece of cake. When exploring how to seamlessly combine different efficient training methods, we that existing Trainers failed to provide a flexible design that would be needed to interleave and inject many different methods into the training loop.  To fulfill this need, we designed a Trainer built for efficiency from first-principles.  In Composer, we carefully designed new abstractions to allow us to have flexibility in all necesssary parts of the training loop.
-
-Composer is designed with **two-way callbacks** ([Howard et al, 2020](https://arxiv.org/abs/2002.04688)) as a first-class citizen. This enables easy injection of efficiency methods throughout the entire training loop.  Not only does the two-way callback system trigger at every part of the training loop, each callback is designed to pass the **entire training state** designed so that any part of training can be modified.
+## 🤔 Why should I use Composer?
```
Comment (ravi-mosaicml) [inline README.md:337]: This section sounds like it would make a great introduction for a conference paper, but I'm not sure whether this needs to be in a readme. Perhaps we can convert this to a short list of bullets, and move towards the top of the readme? I think this should be highlighted and as close to the fold as possible.

**Review by ravi-mosaicml — COMMENTED** [review]
Overall looks good; just some comments throughout

**Review by ravi-mosaicml — APPROVED** [review]
LGTM, though let's address comments later

File: README.md (hunk 7 above)
Comment (hanlint) [reply README.md:207]: The complete list is too long.
